PerimeterXModule/Internals/Validators/PXOriginalTokenValidator.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace PerimeterX
+{
+	public class PXOriginalTokenValidator : PXCookieValidator
+	{
+		public PXOriginalTokenValidator(PxModuleConfigurationSection config) : base(config)
+		{
+		}
+
+		public override bool Verify(PxContext context, IPxCookie pxCookie)
+		{
+			if (pxCookie == null)
+			{
+				return false;
+			}
+
+			if (!pxCookie.Deserialize())
+			{
+				Debug.WriteLine("Failed to decode orignal token", PxConstants.LOG_CATEGORY);
+				context.OriginalTokenError = CALL_REASON_DECRYPTION_FAILED;
+				return false;
+			}
+
+			context.DecodedOriginalToken = pxCookie.DecodedCookie;
+			context.OriginalUUID = pxCookie.Uuid;
+			context.Vid = pxCookie.Vid;
+
+			if (!pxCookie.IsSecured(config.CookieKey, getAdditionalSignedFields(context)))
+			{
+				Debug.WriteLine("Failed to validate original token HMAC", PxConstants.LOG_CATEGORY);
```

Comment:
I know that this task does not handle "aligning the enforcer logs", but if you add _new_ log messages, let's have them at least follow the convention. In this case ([according to the doc](https://docs.google.com/document/d/1A0JX5JcJUsiMmKeYV_YrDqe7z0zr2UYMZyXJSRlq5fo/edit)), it should be something like - 
`Cookie HMAC validation failed, value: *decrypted cookie value*, user-agent: *ua*`

Reply:
not relevant for Original token
a. we dont have UA here
b. we dont return the value right now
not sure its first priority to write this logic